plugins/modules/website.ps1

```diff
@@ -1,27 +1,280 @@
 #!powershell
 
-# Copyright: (c) 2024, Ansible Project
+# Copyright: (c) 2025, Ansible Project
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 #AnsibleRequires -CSharpUtil Ansible.Basic
 
+# Define Bindings Options
+$binding_options = @{
+    type = 'list'
+    elements = 'dict'
+    options = @{
+        ip = @{ type = 'str' }
+        port = @{ type = 'int' }
+        hostname = @{ type = 'str' }
+        protocol = @{ type = 'str' ; default = 'http' ; choices = @('http', 'https') }
+        ssl_flags = @{ type = 'str' ; default = '0' ; choices = @('0', '1', '2', '3') }
+        certificate_hash = @{ type = 'str' ; default = ([string]::Empty) }
+        certificate_store_name = @{ type = 'str' ; default = ([string]::Empty) }
+    }
+}
+
 $spec = @{
     options = @{
         name = @{
             required = $true
-            type = 'str'
+            type = "str"
+        }
+        state = @{
+            type = "str"
+            default = "started"
+            choices = @("absent", "restarted", "started", "stopped")
+        }
+        site_id = @{
+            type = "str"
+        }
+        application_pool = @{
+            type = "str"
+        }
+        physical_path = @{
+            type = "str"
+        }
+        parameters = @{
+            type = "str"
+        }
+        bindings = @{
+            default = @{}
+            type = 'dict'
+            options = @{
+                add = $binding_options
+                set = $binding_options
+                remove = @{
+                    type = 'list'
+                    elements = 'dict'
+                    options = @{
+                        ip = @{ type = 'str' }
+                        port = @{ type = 'int' }
+                        hostname = @{ type = 'str' }
+                    }
+                }
+            }
         }
     }
     supports_check_mode = $true
 }
-
 $module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
 
 $name = $module.Params.name
+$state = $module.Params.state
+$site_id = $module.Params.site_id
+$application_pool = $module.Params.application_pool
+$physical_path = $module.Params.physical_path
+$bindings = $module.Params.bindings
+
+# Custom site Parameters from string where properties are separated by a pipe and property name/values by colon.
+# Ex. "foo:1|bar:2"
+$parameters = $module.Params.parameters
+if ($null -ne $parameters) {
+    $parameters = @($parameters -split '\|' | ForEach-Object {
+            return , ($_ -split "\:", 2)
+        })
+}
```

Comment:
We should remove this format entirely, it was written at a time when modules didn't accept a dictionary. Instead we should do something similar to `web_app_pool`'s `attributes` option which accepts a dictionary value. We probably need to talk a bit more about trying to align this behavour so the option name stays the same and see if we can try and share the code for getting/setting/comparing the values in a module util.